core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -623,28 +629,30 @@ private void seekInternalWithCachingPartiallyReadBlock(long pos) throws IOExcept
     } else {
       mPos = pos / mBlockSize * mBlockSize;
       updateStreams();
-      readCurrentBlockToPos(pos);
+      if (mCurrentCacheStream != null) {
+        cacheCurrentBlockToPos(pos);
+      } else if (mCurrentBlockInStream != null) {
+        seekInternal(pos);
+      } else {
+        Preconditions.checkState(remaining() == 0);
+      }
```

Comment:
I don't think it's possible. The conditions are on the `mCurrentCacheStream` and `mCurrentBlockInStream`, which are set by the `updateStreams()` after the position is updated to the beginning of the block.